docs/docs/data-structures/tree/schema-evolution/compatibility-policies.mdx

```diff
@@ -0,0 +1,208 @@
+There are plans to extend the [`TreeView.compatibility`](../../../api/fluid-framework/treeview-interface#compatibility-property) API with information that allows implementing policies such as this, but the exact API is not finalized.
+
+This policy allows collaboration when the stored schema has additional optional fields that aren't present in the view schema.
+
+### Implementation
+
+```typescript
+// The only types of differences which are tolerated are where the stored schema has optional fields
+// where the view schema has no field.
+function isApplicationLogicCompatible(compatibility: SchemaCompatibilityStatus): boolean {
+	for (const difference of compatibility.differences) {
+		if (difference.mismatch === "nodeType") {
+			return false;
+		}
+
+		for (const fieldDifference of difference.differences) {
+			if (
+				fieldDifference.mismatch !== "fieldKind" ||
+				fieldDifference.view !== undefined ||
+				fieldDifference.stored !== "optional"
+			) {
+				return false;
+			}
+		}
+	}
+
+	return true;
+}
+
+function render(tree: ITree) {
+	const view = tree.viewWith(configuration);
+	const { compatibility } = view;
+	if (!compatibility.canView) {
+		// View schema allows documents that stored schema does not.
+		if (compatibility.canUpgrade) {
+			view.upgradeSchema();
+		} else {
+			// Assuming well-formed rollout, this case should not happen: it means the current view schema
+			// is neither a subset nor a superset of the stored schema.
+			// It could happen in practice if application authors made a change breaking backward-compatibility of the view schema.
+			renderError("This document cannot be opened.");
+			return;
+		}
+	} else if (!isApplicationLogicCompatible(compatibility)) {
+		// Surface to the user that this application version is unable to open this document / updating the application may help.
+		renderError("This version of the application is unable to open the document.");
+		return;
+	}
+
+	view.events.on("schemaChanged", () => {
+		if (!view.compatibility.canView || !isApplicationLogicCompatible(view.compatibility)) {
+			renderError(
+				"Document has been upgraded. This version of the application is unable to open the document.",
+			);
+			return;
+		}
+		// Schema changes invalidate the root of the tree. Real applications would want to react to that here.
+	});
+
+	// Application can open document!
+	renderApplication(view.root);
+}
+```
+
+### Use Cases
+
+- **Applications with frequent optional field additions**
+- **Teams that want faster feature rollout**
+- **Collaborative environments** with mixed client versions
+
+### Trade-offs
+
+**Advantages:**
+- More flexible: Allows optional field evolution
+- Better collaboration: Mixed client versions can work together
+
+**Disadvantages:**
+- Data loss risk: Old clients may lose optional field data during edits e.g. old client constructs a new object for insertion elsewhere in the tree by spread-copying fields from an existing one
```

Comment:
It's not clear what data is being lost and why. This should likely with answers to the following questions: Does the object being spread have data for the new optional field? Will ST ignore that data when View Schema has "allow optional field" enabled and the stored schema says there's a field corresponding to that data?